core/modules/server/routes/put-tiddler.js

```diff
@@ -30,21 +32,29 @@ exports.handler = function(request,response,state) {
 	if(fields.revision) {
 		delete fields.revision;
 	}
+	$tw.perf.timer("parse-tiddler-content");
 	// If this is a skinny tiddler, it means the client never got the full
 	// version of the tiddler to edit. So we must preserve whatever text
 	// already exists on the server, or else we'll inadvertently delete it.
 	if(fields._is_skinny !== undefined) {
+		$tw.perf.timer("get-skinny", "Restore text on server");
 		var tiddler = state.wiki.getTiddler(title);
 		if(tiddler) {
 			fields.text = tiddler.fields.text;
 		}
 		delete fields._is_skinny;
+		$tw.perf.timer("get-skinny");
 	}
+	$tw.perf.timer("add-tiddler", "new Tiddler instance and add to wiki");
 	state.wiki.addTiddler(new $tw.Tiddler(fields,{title: title}));
+	$tw.perf.timer("add-tiddler");
+	$tw.perf.timer("count-change", "Construct etag using changed count of tiddler");
```

Comment:
As @pmario notes, I do think that the performance code is much too intrusive. I think we might have to settle for measuring at the level of individual requests, even though that loses some granularity from what you have here.

But equally, some of these measurements don't make much sense. For example, `getChangeCount()` is really simple and so it's hard to justify measuring it in isolation.

Reply:
I'm not sure which part of the code is time-consuming, so I part them in this granularity.

We can merge some of them, if you think these codes usually don't consume much time. But we should add as much as possible, to get a result like the screenshot above.

> at the level of individual requests

We can already get this information in devtool, about how long a request will take. Server timing is about make this time more clear, by cut it into multiple parts.